governance/api.go

```diff
@@ -112,6 +112,17 @@ func (api *GovernanceKlayAPI) GetRewards(num *rpc.BlockNumber) (*reward.RewardSp
 		return nil, err
 	}
 
+	// mimic legacy reward config cache.
+	// if num is multiple of epoch, don't use ParamsAt(num), but use ParamsAt(num - epoch) instead
+	// see https://github.com/klaytn/klaytn/blob/eabdabed10f7c57ee8809f93dbc0ff67ae9f26bf/reward/reward_config_cache.go#L72-L77
+	epoch := pset.Epoch()
+	if !rules.IsKore && blockNumber%epoch == 0 {
+		pset, err = api.governance.ParamsAt(blockNumber - epoch)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Perhaps factor out as a function in `package reward`?

Reply:
fixed in 57e4716b

Reply:
@blukat29 It can be one of the followings.

(1)
```go
func RewardParamsAt(num uint64, rules params.Rules) (*params.GovParanSet, error)
```
This requires `governanceHelper` inside `RewardDistributor`, which we had chosen to delete.
https://github.com/klaytn/klaytn/blob/c1cb4db027fc2bd08566f7f41cbc45c11dcd3939/reward/reward_distributor.go#L88-L89
We can put it back though.

(2)
```go
func CalcRewardParamBlock(num, epoch uint64, rules params.Rules) (uint64)
```
This will return a number with which the caller can call ParamsAt().

Which do you prefer?

Reply:
I lean towards (2).
But since the mandatory logic `CalcRewardParamBlock` is present in two independent locations Finalize() and api.GetRewards(), please make sure to add test cases to both locations.